DFC.Integration.AVFeed/DFC.Integration.AVFeed.Repository.Sitefinity/CustomApiContextService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using DFC.Integration.AVFeed.Data.Interfaces;
+
+namespace DFC.Integration.AVFeed.Repository.Sitefinity
+{
+    public class CustomApiContextService : HttpClient, ICustomApiContextService
+    {
+        private const string CookieHeaderName = "Set-Cookie";
+        private const string AuthorizationHeaderName = "Authorization";
+        private const string BearerHeaderName = "Bearer";
+        private static readonly Uri ClearRequestUrl = new Uri(ConfigurationManager.AppSettings.Get("Sitefinity.ClearAVBinURL"));
+        private static string BaseAddressUrl => ConfigurationManager.AppSettings.Get("Sitefinity.BaseAddress");
+        private static string AuthCookieEndpoint => ConfigurationManager.AppSettings.Get("Sitefinity.AuthCookieEndpoint");
+
+        private readonly CookieContainer cookieContainer;
+        private readonly IApplicationLogger applicationLogger;
+        private readonly ITokenClient tokenClient;
+
+        public CustomApiContextService(IApplicationLogger applicationLogger, ITokenClient tokenClient)
+        {
+            this.applicationLogger = applicationLogger;
+            this.tokenClient = tokenClient;
+            BaseAddress = new Uri(BaseAddressUrl);
+            cookieContainer = new CookieContainer();
+        }
+
+        private async Task ConfigureCoookies()
+        {
+            var accessToken = await tokenClient.GetAccessTokenAsync();
+
+            DefaultRequestHeaders.Add(AuthorizationHeaderName, $"{BearerHeaderName} {accessToken}");
+
+            applicationLogger.Info($"Start - Auth Cookie endpoint {AuthCookieEndpoint} called.");
+
+            var response = await GetAsync(new Uri(AuthCookieEndpoint));
+            response.Headers.TryGetValues(CookieHeaderName, out var cookies);
+            var cookieList = cookies as IList<string> ?? cookies.ToList();
+            if (!cookieList.Any())
+            {
+                applicationLogger.Info(
+                    $"Auth Cookie client {AuthCookieEndpoint} called failed with error {response.StatusCode}.");
+                throw new ApplicationException("Couldn't get auth cookie token. Error: " + response.StatusCode);
+            }
+            
+            cookieContainer.SetCookies(BaseAddress, string.Join(", ", cookieList));
+        }
+
+        public async Task ClearAVsRecycleBinAsync(int itemCount)
```

Comment:
- This doesn't clear / empty recycle bin, so the method name is misleading.
- Don't you want to indicate there are more content to delete?

Reply:
Changed name to indicate we are only clearing some of the records.